### PR TITLE
DM-13270: Cleanup ccdImage and gtransfo APIs

### PR DESCRIPTION
**Pull request by parejkoj**
* Rename sky2TP->skyToTangentPlane
* Rename other ccdImage tangent plane/pixel methods and members: `X2Y`->`XToY`
  and cleanup TP->TangentPlane, CTP->CommonTangentPlane, etc.

Rename gtransfo subclasses:
 * TanPix2RaDec->TanPixelToRaDec
 * TanRaDec2Pix->TanRaDecToPixel
 * TanSipPix2RaDec->TanSipPixelToRaDec

### CODE REVIEW

File: include/lsst/jointcal/Gtransfo.h [hunk 1]
```diff
@@ -579,55 +579,57 @@ class BaseTanWcs : public Gtransfo {
...
+    //! pixToTan describes the transfo from pix to tangent plane (degrees). TangentPoint in degrees.
     //! Corrections are applied between Lin and deprojection parts (as in Swarp).
-    TanPix2RaDec(GtransfoLin const &pix2Tan, Point const &tangentPoint,
-                 const GtransfoPoly *corrections = nullptr);
+    TanPixelToRaDec(GtransfoLin const &pixToTan, Point const &tangentPoint,
+                    const GtransfoPoly *corrections = nullptr);
 
     //! the transformation from pixels to tangent plane (degrees)
-    GtransfoPoly getPix2TangentPlane() const;
+    GtransfoPoly getPixToTangentPlane() const;
```
Comment (czwa) [inline include/lsst/jointcal/Gtransfo.h]: `getPixelToTangentPlane()`?

File: include/lsst/jointcal/Gtransfo.h [hunk 2]
```diff
@@ -639,23 +641,24 @@ class TanPix2RaDec : public BaseTanWcs {
...
+    //! pixToTan describes the transfo from pix to tangent plane (degrees). TangentPoint in degrees.
     //! Corrections are applied before Lin.
-    TanSipPix2RaDec(GtransfoLin const &pix2Tan, Point const &tangentPoint,
-                    const GtransfoPoly *corrections = nullptr);
+    TanSipPixelToRaDec(GtransfoLin const &pixToTan, Point const &tangentPoint,
+                       const GtransfoPoly *corrections = nullptr);
 
     //! the transformation from pixels to tangent plane (degrees)
-    GtransfoPoly getPix2TangentPlane() const;
+    GtransfoPoly getPixToTangentPlane() const;
```
Comment (czwa) [inline include/lsst/jointcal/Gtransfo.h]: `getPixelToTangentPlane()`?

File: include/lsst/jointcal/Gtransfo.h [hunk 3]
```diff
@@ -579,55 +579,57 @@ class BaseTanWcs : public Gtransfo {
 
     //! Get a transform from pixels to tangent plane (degrees)
     //! This is a linear transform plus the effects of the correction
-    virtual GtransfoPoly getPix2TangentPlane() const = 0;
+    virtual GtransfoPoly getPixToTangentPlane() const = 0;
```
Comment (czwa) [inline include/lsst/jointcal/Gtransfo.h]: `getPixelToTangentPlane()`?

File: src/Gtransfo.cc [hunk 4]
```diff
@@ -1300,10 +1300,10 @@ static double deg2rad(double degree) { return degree * M_PI / 180.; }
 static double rad2deg(double rad) { return rad * 180. / M_PI; }
 
 /*************  WCS transfo ******************/
-/************** LinPix2Tan *******************/
+/************** LinPixToTan *******************/
```
Comment (czwa) [inline src/Gtransfo.cc]: `LinPixelToTan` here and throughout?

**Comment by parejkoj** [comment]
Thanks: this was done in a series of steps as I decided on better and better names for things (and then flattend into one commit), but I clearly didn't catch everything at each step of the way. I'm glad you caught those.

**Pull request merged**